src/plugins/discover/public/application/main/components/log_explorer/log_explorer_main_route.tsx

```diff
@@ -0,0 +1,240 @@
+        const ipList = ip.list as Array<SavedObject<DataViewAttributes>>;
+        const indexPatternData = resolveIndexPattern(ip, searchSource, toastNotifications);
+
+        setIndexPatternList(ipList);
+
+        return indexPatternData;
+      } catch (e) {
+        setError(e);
+      }
+    },
+    [config, data.dataViews, history, isDev, toastNotifications]
+  );
+
+  const loadSavedSearch = useCallback(async () => {
+    try {
+      const currentSavedSearch = await getSavedSearch(id, {
+        search: services.data.search,
+        savedObjectsClient: core.savedObjects.client,
+        spaces: services.spaces,
+      });
+
+      const loadedIndexPattern = await loadDefaultOrCurrentIndexPattern(
+        currentSavedSearch.searchSource
+      );
+
+      if (!loadedIndexPattern) {
+        return;
+      }
+
+      if (!currentSavedSearch.searchSource.getField('index')) {
+        currentSavedSearch.searchSource.setField('index', loadedIndexPattern);
+      }
+
+      setSavedSearch(currentSavedSearch);
+
+      if (currentSavedSearch.id) {
+        chrome.recentlyAccessed.add(
+          getSavedSearchFullPathUrl(currentSavedSearch.id),
+          currentSavedSearch.title ?? '',
+          currentSavedSearch.id
+        );
+      }
+    } catch (e) {
+      if (e instanceof DataViewSavedObjectConflictError) {
+        setError(e);
+      } else {
+        redirectWhenMissing({
+          history,
+          navigateToApp: core.application.navigateToApp,
+          basePath,
+          mapping: {
+            search: '/',
+            'index-pattern': {
+              app: 'management',
+              path: `kibana/objects/savedSearches/${id}`,
+            },
+          },
+          toastNotifications,
+          onBeforeRedirect() {
+            getUrlTracker().setTrackedUrl('/');
+          },
+          theme: core.theme,
+        })(e);
+      }
+    }
+  }, [
+    id,
+    services.data.search,
+    services.spaces,
+    core.savedObjects.client,
+    core.application.navigateToApp,
+    core.theme,
+    loadDefaultOrCurrentIndexPattern,
+    chrome.recentlyAccessed,
+    history,
+    basePath,
+    toastNotifications,
+  ]);
+
+  const onDataViewCreated = useCallback(
+    async (dataView: unknown) => {
+      if (dataView) {
+        setShowNoDataPage(false);
+        setError(undefined);
+        await loadSavedSearch();
+      }
+    },
+    [loadSavedSearch]
+  );
+
+  useEffect(() => {
+    loadSavedSearch();
+  }, [loadSavedSearch]);
+
+  useEffect(() => {
+    chrome.setBreadcrumbs(
+      savedSearch && savedSearch.title
+        ? getSavedSearchBreadcrumbs(savedSearch.title)
+        : getRootBreadcrumbs()
+    );
+  }, [chrome, savedSearch]);
+
+  if (showNoDataPage) {
+    const analyticsServices = {
+      coreStart: core,
+      dataViews: {
+        ...data.dataViews,
+        hasData: {
+          ...data.dataViews.hasData,
+
+          // We've already called this, so we can optimize the analytics services to
```

Comment:
In general Discover seems to have more granular (and probably much more meaningful) analytics. We probably want to bear this in mind for functionality we add.